src/App.vue

```diff
@@ -12,37 +13,113 @@ import WidgetPicto from './components/WidgetPicto.vue';
 
 const isSettingsOpen = ref(false);
 
+function sleep(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
 const widgets = [
-  WidgetWeather,
-  WidgetNews,
-  WidgetAtCoder,
-  WidgetCalendar,
-  WidgetClock
+  { name: 'WidgetWeather', component: WidgetWeather },
+  { name: 'WidgetNews', component: WidgetNews },
+  { name: 'WidgetAtCoder', component: WidgetAtCoder },
+  { name: 'WidgetCalendar', component: WidgetCalendar },
+  { name: 'WidgetClock', component: WidgetClock }
 ];
 
+const slideInterval = ref<number | undefined>(undefined);
 const widgetIndex = ref(0);
+const direction = ref(0);
+
+function startAutoSlide() {
+  stopAutoSlide();
+  slideInterval.value = setInterval(() => {
+    nextWidget();
+  }, 10000) as unknown as number;
+}
+
+function stopAutoSlide() {
+  if (slideInterval.value) {
+    clearInterval(slideInterval.value);
+    slideInterval.value = undefined;
+  }
+}
 
 function nextWidget() {
   widgetIndex.value = (widgetIndex.value + 1) % widgets.length;
+  direction.value = 0;
 }
 
 function prevWidget() {
   widgetIndex.value = (widgetIndex.value - 1 + widgets.length) % widgets.length;
+  direction.value = 1;
 }
 
-nextWidget();
-setInterval(nextWidget, 10000);
+async function setWidget(widgetName: String) {
```

Comment:
```suggestion
async function setWidget(widgetName: ScrollTarget) {
```

文字列だとしても`String`じゃなくて`string`を使おう
https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#the-primitives-string-number-and-boolean